webapp/src/main/java/com/box/l10n/mojito/monitoring/PrefixHierarchicalNameMapper.java

```diff
@@ -1,45 +1,74 @@
 package com.box.l10n.mojito.monitoring;
 
-import com.google.common.base.Preconditions;
+
 import com.google.common.base.Strings;
+
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nullable;
+
 import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.Tag;
 import io.micrometer.core.instrument.config.NamingConvention;
 import io.micrometer.core.instrument.util.HierarchicalNameMapper;
 
-import java.util.stream.Collectors;
+import static com.google.common.base.Preconditions.checkNotNull;
 
 public class PrefixHierarchicalNameMapper implements HierarchicalNameMapper {
 
     String namePrefix;
     String tagKeyPrefix;
+    String stripCharacters;
     NamingConvention nameConventionOverride;
 
-    public PrefixHierarchicalNameMapper(String namePrefix, String tagKeyPrefix, NamingConvention nameConventionOverride) {
-        Preconditions.checkNotNull(namePrefix, tagKeyPrefix, nameConventionOverride);
-        this.namePrefix = namePrefix;
-        this.tagKeyPrefix = tagKeyPrefix;
+    public PrefixHierarchicalNameMapper(String namePrefix,
+                                        String tagKeyPrefix,
+                                        String stripCharacters,
+                                        @Nullable NamingConvention nameConventionOverride) {
+        this.namePrefix = checkNotNull(namePrefix);
+        this.tagKeyPrefix = checkNotNull(tagKeyPrefix);
+        this.stripCharacters = checkNotNull(stripCharacters);
         this.nameConventionOverride = nameConventionOverride;
     }
 
+    public PrefixHierarchicalNameMapper(String namePrefix,
+                                        String tagKeyPrefix,
```

Comment:
why do we need multiple ctor?

Reply:
We don't need them, but it makes easier to test 😄

Reply:
yeah I saw that later reading the test. I tend to prefer more verbose test for a cleaner class (if the ctor is not used in the actual code).

There is a cool feature in intellij to add parameters to functions, it just updates everything at once with default value. It kind of alleviate the need of default values in cto

Reply:
Updated